src/lib/core/coio.c

```diff
@@ -205,8 +210,8 @@ coio_connect_timeout(const char *host, const char *service, int host_hint,
 	diag_set(SocketError, sio_socketname(fd), "connection failed");
 	return fd;
 out:
-	if (host_hint == 0)
-		freeaddrinfo(ai);
+	if (ai_resolve != NULL)
+		freeaddrinfo(ai_resolve);
 	else
 		free(ai_local.ai_addr);
```

Comment:
Nit: Better initialize both `ai_local` and `ai_resolve` to `NULL` and free both of them here if not `NULL`.

Reply:
Done.

```diff
@ -170,7 +170,7 @@ coio_connect_timeout(const char *host, const char *service, int h>

        struct addrinfo *ai;
        struct addrinfo *ai_resolve = NULL;
-       struct addrinfo ai_local;
+       struct addrinfo ai_local = { .ai_addr = NULL };
        if (host != NULL && service != NULL && host_hint != 0) {
                if (coio_fill_addrinfo(&ai_local, host, service,
                                       host_hint) != 0)
@@ -212,7 +212,7 @@ coio_connect_timeout(const char *host, const char *service, int h>
 out:
        if (ai_resolve != NULL)
                freeaddrinfo(ai_resolve);
-       else
+       if (ai_local.ai_addr != NULL)
                free(ai_local.ai_addr);
        return fd;
```